include/nlohmann/detail/input/json_sax.hpp

```diff
@@ -227,19 +244,18 @@ class json_sax_dom_parser
         return true;
     }
 
+    HANDLE_START_END_POS_DEFINITION(start_object,
+    {
+        reinterpret_cast<::nlohmann::detail::json_base_class_with_start_end_markers*>(ref_stack.back())->start_position = m_lexer_ref->get_position() - 1;
```

Comment:
Due to the SFINAE changes, the `reinterpret_cast<::nlohmann::detail::json_base_class_with_start_end_markers*>()` is not needed.